contract_sale_invoicing/models/contract.py

```diff
@@ -9,23 +9,141 @@ class ContractContract(models.Model):
+                        so_domain, order='id asc')
+                    sale_order_line_product_qty = {}
+                    for order_id in order_ids:
+                        if not order_id.order_line.mapped('invoice_lines'):
+                            for line in order_id.order_line:
+                                if sale_order_line_product_qty.\
+                                        get(line.product_id.id):
+                                    sale_order_line_product_qty[
+                                        line.product_id.id
+                                    ] += line.qty_to_invoice
+                                else:
+                                    sale_order_line_product_qty[
+                                        line.product_id.id
+                                    ] = line.qty_to_invoice
+                                if invoice_line.get(
+                                    'product_id'
+                                ) in sale_order_line_product_qty:
+                                    if sale_order_line_product_qty.get(
+                                            line.product_id.id
+                                    ) > invoice_line.get('quantity'):
+                                        remain_qty = \
+                                            sale_order_line_product_qty.\
+                                            get(invoice_line.get('product_id')
+                                                ) - invoice_line.get(
+                                                'quantity') or 0
+                                        invoice_line_list.append({
+                                            'uom_id':
+                                            contract_line_rec.uom_id.id,
+                                            'product_id':
+                                            invoice_line.get('product_id'),
+                                            'quantity': remain_qty or 0,
+                                            'discount':
+                                            contract_line_rec.discount,
+                                            'contract_line_id':
+                                            contract_line_rec.id,
+                                            'name': contract_line_rec.name,
+                                            'account_analytic_id':
+                                            contract_line_rec.
+                                            analytic_account_id.id,
+                                            'sale_line_ids':
+                                            [(6, 0, [line.id])],
+                                            'price_unit':
+                                            contract_line_rec.price_unit,
+                                            'account_id':
+                                            order_id.partner_id.
+                                            property_account_receivable_id.id,
```

Comment:
@Nikul-Chaudhary @murtuzasaleh You need to use the income account of the product category, not the receivable account of the partner...

Reply:
You also need to apply the fiscal position (as it may change the account)